lib/Service/OnedriveAPIService.php

```diff
@@ -156,6 +135,7 @@ public function request(string $userId, string $endPoint, array $params = [], st
 					'Authorization' => 'bearer ' . $accessToken,
 					'User-Agent' => 'Nextcloud OneDrive integration'
```

Comment:
The user agent is invalid [according to RFC9110](https://www.rfc-editor.org/rfc/rfc9110#name-user-agent)


> User-Agent = product *( RWS ( product / comment ) )
> product         = token ["/" product-version]
> product-version = token
> tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "." /
 "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
> token = 1*tchar


(ALPHA being A-Z a-z)

a better value would be something like "nextcloud/nextCloudVersion nextcloud-onedrive/pluginversion"

Additionally, this should be set by a middleware handler as well to get it out of the control logic. Kiota has a [Telemetry Handler](https://github.com/microsoft/kiota-http-guzzle-php/blob/main/src/Middleware/TelemetryHandler.php) that can be used to do that.

Not saying this is blocking for this PR, we should probably create a separate issue for that.

Reply:
Thanks for letting me know about this, I created a separate issue so as not to forget about this problem.

And thanks again for the suggestion on how to fix the problem using KiotaMiddleware